referenceframe/frame_test.go

```diff
@@ -280,3 +281,90 @@ func TestFrame(t *testing.T) {
 	test.That(t, err, test.ShouldBeNil)
 	test.That(t, sFrame, test.ShouldResemble, expStaticFrame)
 }
+
+func TestFrameToJSONAndBack(t *testing.T) {
+	// static frame
+	static, err := NewStaticFrame("foo", spatial.NewPose(r3.Vector{X: 1, Y: 2, Z: 3}, &spatial.R4AA{Theta: math.Pi / 2, RX: 4, RY: 5, RZ: 6}))
+	test.That(t, err, test.ShouldBeNil)
+
+	jsonData, err := frameToJSON(static)
+	test.That(t, err, test.ShouldBeNil)
+
+	static2, err := jsonToFrame(json.RawMessage(jsonData))
+	test.That(t, err, test.ShouldBeNil)
+
+	test.That(t, static.Name(), test.ShouldResemble, static2.Name())
+
+	staticF, ok := static.(*staticFrame)
+	test.That(t, ok, test.ShouldBeTrue)
+
+	staticF2, ok := static2.(*staticFrame)
+	test.That(t, ok, test.ShouldBeTrue)
+
+	test.That(t, spatial.PoseAlmostCoincident(staticF.transform, staticF2.transform), test.ShouldBeTrue)
+
+	staticFrame, ok := static.(*staticFrame)
+	test.That(t, ok, test.ShouldBeTrue)
+	tailGeoFrame := tailGeometryStaticFrame{staticFrame: staticFrame}
```

Comment:
can we write these tests with either test.Resembles or add some helper function / method that allows us to answer the question whether two frames are (semantically) equal?

Same goes for the entire frame system.

If we have to walk 2 frame system across all frames to determine if they are equal that is going to be very inconvenient.